Rename public functions and stop exporting several public API methods (#699)
JuliaIO/HDF5.jl: docs/src/index.md

> t = t_open(parent, name)

Reviewer: ```suggestion
t = open_datatype(parent, name)
```
(and similar for previous three lines. For some reason I can't get Github to let me add a suggestion for those lines...)

```

> These open the named group, dataset, attribute, and committed HDF5.datatype, respectively. For datasets, `apl` stands for "access parameter list" and provides opportunities for more sophisticated control (see the [HDF5](https://www.hdfgroup.org/solutions/hdf5/) documentation).

Reviewer: ```suggestion
These open the named group, dataset, attribute, and committed datatype, respectively. For datasets, `apl` stands for "access parameter list" and provides opportunities for more sophisticated control (see the [HDF5](https://www.hdfgroup.org/solutions/hdf5/) documentation).
```

Extra find/replace?

New objects can be created in the following ways:
```julia
g = g_create(parent, name[, lcpl, dcpl])
dset = HDF5.create_dataset(parent, name, data[, lcpl, dcpl, dapl])
attr = a_create(parent, name, data)
```

> creates groups, datasets, and attributes without writing any data to them. You can then use `write(obj, data)` to store the data. The optional property lists allow even more fine-grained control. This syntax uses `data` to infer the object's "HDF5.datatype" and "HDF5.dataspace"; for the most explicit control, `data` can be repl mainaced with `dtype, dspace`, where `dtype` is an `HDF5Datatype` and `dspace` is an `HDF5Dataspace`.

Reviewer: ```suggestion
creates groups, datasets, and attributes without writing any data to them. You can then use `write(obj, data)` to store the data. The optional property lists allow even more fine-grained control. This syntax uses `data` to infer the object's "HDF5.datatype" and "HDF5.dataspace"; for the most explicit control, `data` can be replaced with `dtype, dspace`, where `dtype` is an `HDF5Datatype` and `dspace` is an `HDF5Dataspace`.
```
